### DISCUSSION TITLE
How to setup contact page with multi language mode?

### DISCUSSION THREAD

**Discussion by ifrx** [datenstrom/community#230] [Ask a question] [answered]
This issue might belong to [yellow-plugins…contact](https://github.com/datenstrom/yellow-plugins/tree/master/contact) but I couldn't find an Issues section there.

Hi @markseu
I've setup a simple Yellow test website (using PHP Version 7.0.15) with [/content/1-home], […], pasted contact.zip into [/system/plugins/] and all worked nicely with [/content/contact].

But, after switching on Multi language mode via [/system/config/config.ini] and with [/content/1-de/1-home], […], and [/content/2-en/1-home], and keeping [/content/contact], Yellow produced "File not found" on URL/contact. Apart from that, Multi language mode was working as expected.

So, I thought I should configure [/system/config/config.ini] "ContactLocation = contact page location" as indicated in [yellow-plugins…contact](https://github.com/datenstrom/yellow-plugins/tree/master/contact). Tried "ContactLocation: /content/contact/" also "ContactLocation: /contact/" both without success.

Finally, I switched back to "Navigation: navigation" and "MultiLanguageMode: 0" via [/system/config/config.ini] and with [/content/1-home], […], keeping [/content/contact] and URL/contact was working again.

How should "ContactLocation = contact page location" look like in [/system/config/config.ini]?
What would be the right way to setup Contact plugin in Multi language mode?

**Comment by markseuffert** [comment, marked as answer]
Hi ifrx,

you found the right place. The multi language mode is a bit difficult to set up. After you switched it on you have to do some adjustments in your content folder. The contact plugin will create a folder inside the content folder, but when the multi language mode is switched on content must be moved one level down inside the language folders.

Here's an example how it should look like:

![screen shot 2017-02-12 at 11 15 39](https://cloud.githubusercontent.com/assets/2429165/22861290/bd535be0-f115-11e6-92e3-b576534af940.png)

Hope this helps. 😄

**Comment by ifrx** [comment]
Hi @markseu 

"/content/LANG/contact" is working now. Was too simple in that regard ;-) Thank you!

Just one thing is left:

I have added a [contact] shortcut to another page per language as in [How to use a contact page](https://github.com/datenstrom/yellow-plugins/tree/master/contact#how-to-use-a-contact-page).
The form shows up nicely but on "send" it went to "URL/contact" instead of going to "URL/LANG/contact" and said "File not found …" again.

So, tried to configure "ContactLocation = contact page location" in "/system/config/config.ini" as indicated in [How to configure a contact page](https://github.com/datenstrom/yellow-plugins/tree/master/contact#how-to-configure-a-contact-page).

Tried
```
ContactLocation: /de/contact/
ContactLocation: /en/contact/
```
and vice versa. But this would always process the last ContactLocation line on "send".
That is better than "File not found …" but is still not a solution.

Do you have an example on how to configure "ContactLocation = contact page location" in "config.ini", so that a [contact] shortcut of a given page will be processed according to the relevant language like "URL/LANG/contact"?

**Comment by markseuffert** [comment]
Yes, that's possible by adding the location to the shortcut:

```[contact /en/contact/]``` in an English page
```[contact /de/contact/]``` in a German page

This will override the configuration file. Would that work for you?

**Comment by ifrx** [comment]
That is working brilliantly!

Would it be possible to insert another section after [How to configure a contact page](https://github.com/datenstrom/yellow-plugins/tree/master/contact#how-to-configure-a-contact-page)?
Maybe that way:

## How to configure a contact page in multi language mode?

To enable Yellow in [multi language mode](https://developers.datenstrom.se/help/language-configuration#multi-language-mode) to process the contact form in the right language, move your `/contact` folder into each language subfolder.

If you are additionally using a `[contact]` shortcut in a page of a language subfolder, write the web location of the related `/contact` folder into the shortcut, eg `[contact /en/contact/]`.

**Comment by markseuffert** [comment]
Thanks for your suggestion. Maybe we can improve the installation, so that all plugins work with the multi language mode. The software installation is a new addition to Yellow, so there are some things that can be done better. I added a note to the internal TODO list. 

See also datenstrom/community#187.

**Comment by ifrx** [comment]
Have added a comment to datenstrom/community#187. Thank you
